### PR TITLE
Rename command-line options for VC randomization

### LINKED ISSUE #581: Early detection of butterfly effect
Boogie should provide robust tooling for "fuzzing" proved programs so that an unstable proof (due to butterfly effect) is revealed early potentially before verified code is checked in.

cc: @keyboardDrummer

### PR DESCRIPTION
**Pull request by shazqadeer**
Closes #581

### CODE REVIEW

**Review by atomb — CHANGES_REQUESTED** [review]
Overall, I think this is a great improvement. Given that there's some code out there that depends on the old option name, I'm wondering whether it would be reasonable to still accept it.

File: Source/ExecutionEngine/CommandLineOptions.cs [hunk 1]
```diff
@@ -1450,9 +1450,9 @@ protected override bool ParseOption(string name, CommandLineParseState ps)
           ps.GetIntArgument(x => VcsCores = x, a => 1 <= a);
           return true;
 
-        case "randomSeedIterations":
```
[the remark this answers is not in the document]
Comment (shazqadeer) [reply Source/ExecutionEngine/CommandLineOptions.cs:1453]: I assume you are talking about the Dafny dependency on Boogie.  What process do you suggest we follow for deprecating Boogie features that are no longer necessary or covered by other features? Specifically, when would it be reasonable to drop ```-randomSeedIterations" entirely.

[the remark this answers is not in the document]
Comment (keyboardDrummer) [reply Source/ExecutionEngine/CommandLineOptions.cs:1453]: Dafny doesn't use the Boogie CLI, so there's no Dafny dependency on this CLI option name.

[the remark this answers is not in the document]
Comment (atomb) [reply Source/ExecutionEngine/CommandLineOptions.cs:1453]: For anyone using the new top-level commands, that's true. But for anyone using the older options, they're passed through to Boogie directly from the command line. Even though Dafny is using Boogie as a library, it still uses Boogie's option parser for that case.

I'm not sure I have a concrete answer yet as to when it'd be inconsequential to drop it, but I'd like to at least update the Dafny documentation before Boogie changes.

We're overall moving away from the model of automatically passing Boogie options through verbatim, so this isn't likely to be as much of an issue in the future.

[the remark this answers is not in the document]
Comment (shazqadeer) [reply Source/ExecutionEngine/CommandLineOptions.cs:1453]: I added a case for the old option name. I would like to request you to update Dafny documentation soon and then circle back with a Boogie PR to remove the old option name. Thanks for your feedback and for the great work you did in adding this feature to Boogie.

**Review by atomb — APPROVED** [review]

**Pull request merged**